interview-questions.md

```diff
@@ -8,36 +8,65 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-Your answer:
 
-Researched answer:
 
 3. Name three rails generator commands. What is created by each?
 
-Your answer:
+Your answer: You have rails generate model. This rails generate command builds a model inside a rails application for us with us adding the parameters for it like $rails generate model Animal name:string age:integer dob:date. Another generator command is rails generate migration.  Rails migration is a way to add columns to a table.  This comes in handy becuase you can add new columns or add columns that you forgot to like I did in one of my projects so I had to utilize rails generate migration and add the two columns I needed. The last rails generator command I can think of is rails generate resources.  This command is my favorite since it essentially has you build a model like rails generate model but it also adds the routes you need inside the rails project so you do not have to manually do them yourself.
 
-Researched answer:
+Researched answer: Afte some research I came accross pretty much what I said with just a little more detail.  Instead of touching on them again I want to talk about he one I didn't mention above and that is rails generate controller.  When you run this it add all the files associated with the name of the controller you chose.  It then allows you to to be able to access the file and add the methods you want to use.
 
 4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
 
 action: "GET" location: /students
+You would use the method index to access all the data that has been stored in table.  
 
 action: "POST" location: /students
+For this route you use the create method.  This method lets you create a new instance for the table.
 
 action: "GET" location: /students/new
+This will help create a new instance of itself by creating the html for you will need to make a new instance. This will use the new method.
 
 action: "GET" location: /students/2
+This will use the show method. This allows you to you look at a particular instance whose primary key matches the number you put in the URL.
 
 action: "GET" location: /students/2/edit
+You will use the edit method here which gets the html form so that edit can use.  This will have you edit one particular instance with the udpate method.
 
 action: "PATCH" location: /students/2
+This uses the update method and this with grab the instance that has the primary key 2 and help you update the data in the instance.
 
 action: "DELETE" location: /students/2
+You would use the delete method.  The delete method deletss the instance you declare and completely removes it from the table.
 
 5. As a developer, you are making an application to manage your to do list. Create 10 user stories that will help you get your application started. Read more about [user stories](https://www.atlassian.com/agile/project-management/user-stories).
+
```

Comment:
The idea of user stories is to share the user experience with your application. For instance, 
"As a developer I want to be able to have a name, and projected duration of the task, and to be able to determine if the task has been completed or not"
User story:
`As a user, I see a form that allows me to create a task by entering a name and projected duration.`
`As a user, I can create a task.`
`As a user, I can mark a task completed or not completed.`